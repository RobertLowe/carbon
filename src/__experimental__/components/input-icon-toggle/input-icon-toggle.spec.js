import React from 'react';
import TestRenderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import Icon from 'components/icon';
import { assertStyleMatch } from '../../../__spec_helper__/test-utils';
import ValidationIcon from '../../../components/validations/validation-icon.component';
import InputIconToggleStyle from './input-icon-toggle.style';
import BaseTheme from '../../../style/themes/base';
import InputIconToggle from './input-icon-toggle.component';
import StyledIcon from '../../../components/icon/icon.style';


describe('InputIconToggle', () => {
  describe('when initiated with the disabled prop set to true', () => {
    it('does not render anything', () => {
      expect(render({ disabled: true }).isEmptyRender()).toBeTruthy();
    });
  });

  describe('when initiated with the readOnly prop set to true', () => {
    it('does not render anything', () => {
      expect(render({ readOnly: true }).isEmptyRender()).toBeTruthy();
    });
  });

  describe('when initiated without children', () => {
    it('renders an Icon component with an icon type that was specified in the props', () => {
      expect(render({ inputIcon: 'settings' }).contains(<Icon type='settings' />)).toBeTruthy();
    });
  });

  describe('when initiated with children', () => {
    it('renders as expected', () => {
      expect(render({ children: 'mock content' }, TestRenderer.create)).toMatchSnapshot();
    });
  });

  describe.each(['error', 'warning', 'info'])('when %s validation prop is string', (validationProp) => {
    it('renders a validation icon', () => {
      const wrapper = render({ [validationProp]: 'Message' });
      const validationIcon = wrapper.find(ValidationIcon);
      expect(validationIcon.exists()).toBe(true);
    });
  });

  describe.each(['error', 'warning', 'info'])('when %s validation prop is true', (validationProp) => {
    it('renders colored input icon when it is provided', () => {
      const wrapper = render({ inputIcon: 'dropdown', [validationProp]: true }, mount);
      const iconStyle = wrapper.find(InputIconToggleStyle);

      assertStyleMatch({
        color: BaseTheme.colors[validationProp]
      }, iconStyle, { modifier: `${StyledIcon}:before` });
    });
  });

  describe('sizes', () => {
    [['small', '32px'], ['medium', '40px'], ['large', '48px']].forEach((size) => {
      it(`updates the width for ${size[0]}`, () => {
        assertStyleMatch({
          width: size[1]
        }, render({ size: size[0] }, TestRenderer.create).toJSON());
      });
    });
  });
});

function render(props, renderer = shallow) {
  const defaultProps = { inputIcon: 'settings' };

  return renderer(
    <InputIconToggle { ...defaultProps } { ...props } />
  );
}
