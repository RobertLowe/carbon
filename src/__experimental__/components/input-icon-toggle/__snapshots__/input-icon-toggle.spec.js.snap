// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputIconToggle when initiated with children renders as expected 1`] = `
.c1 {
  display: inline-block;
  position: relative;
  color: rgba(0,0,0,0.65);
  background-color: transparent;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 24px;
  width: 24px;
}

.c1:hover {
  color: rgba(0,0,0,0.90);
  background-color: transparent;
}

.c1::before {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: CarbonIcons;
  content: "\\e91a";
  font-size: 16px;
  font-style: normal;
  font-weight: normal;
  line-height: 16px;
  vertical-align: middle;
  display: block;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: -11px;
  width: 40px;
}

<span
  className="c0"
  size="medium"
>
  <span
    className="c1"
    data-component="icon"
    data-element="settings"
    disabled={false}
    fontSize="small"
    type="settings"
  />
</span>
`;
